server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jdkim/MimeMessageHeaders.java

```diff
@@ -0,0 +1,71 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets.jdkim;
+
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.james.jdkim.api.Headers;
+
+/**
+ * An adapter to let DKIMSigner read headers from MimeMessage
+ */
+final class MimeMessageHeaders implements Headers {
+
+    private final Map<String, List<String>> headers;
+    private final List<String> fields;
+
+    @SuppressWarnings("unchecked")
+    public MimeMessageHeaders(MimeMessage message)
+            throws MessagingException {
+        headers = new HashMap<String, List<String>>();
+        fields = new LinkedList<String>();
+        for (Enumeration<String> e = message.getAllHeaderLines(); e
+                .hasMoreElements();) {
+            String head = e.nextElement();
+            int p = head.indexOf(':');
+            if (p <= 0)

```

Comment:
Moving is one issue, refactoring is an other. Do we want to take time doing this?

(It is an external contribution, we can ask Shrumit to do it...)